app/listViewComponent.js

```diff
@@ -0,0 +1,307 @@
+    };
+  }
+
+  // viewDidLoad
+  componentDidMount() {
+
+    AppState.addEventListener("change", this._handleAppStateChange);
+    this._handleAppStateChange(AppState.currentState);
+
+    firestack.analytics.logEventWithName("pageView", {
+      screen: "ListViewComponent"
+    });
+    firestack.analytics.logEventWithName("openListView");
+  }
+
+  // viewDidUnload
+  componentWillUnmount() {
+    AppState.removeEventListener("change", this._handleAppStateChange);
+    Database.stopListening();
+  }
+
+  _handleAppStateChange = appState => {
+    if (appState == "active") {
+      // viewDidAppear
+      console.log("Start listening to other users");
+      Database.listenToUsers(this.onOtherUserUpdatedLocation);
+    } else if (appState == "inactive" || appState == "background") {
+      // viewDidDisappear
+      console.log("Stop listening to other users");
+      Database.stopListening();
+    }
+  };
+
+  // This is called with lat & lng being nil if a marker gets removed
+  onOtherUserUpdatedLocation = (
+    userId,
+    lat,
+    lng,
+    timestamp,
+    twitterUsername,
+    shouldSetState
+    ) => {
+    // if (userId == this.props.userId) {
+    //   return; // We don't want to show ourselve, as it might cover other people
+    // }
+
+    let foundExisting = -1;
+    let coordinatesProvided = !(lat == null && lng == null);
+    let coordinate = null;
+    let description = timeDifference(new Date(), timestamp);
+    let userDistance = 0;
+
+    if (coordinatesProvided) {
+      coordinate = { latitude: parseFloat(lat), longitude: parseFloat(lng) };
+      userDistance = this._calculateDistance(coordinate);
+    }
+
+    for (let i = 0; i < this.state.markers.length; i++) {
+      if (this.state.markers[i]["key"] == userId) {
+        if (coordinatesProvided) {
+          this.state.markers[i]["coordinate"] = coordinate;
+          this.state.markers[i]["distance"] = userDistance;
+          this.state.markers[i]["description"] = description;
+        }
+        foundExisting = i;
+      }
+    }
+
+    if (foundExisting > 0 && !coordinatesProvided) {
+      // we have to remove this marker from our list
+      // as the user disabled their location sharing
+      console.log("Removing the marker here");
+      this.state.markers.splice(foundExisting, 1);
+    }
+
+    // This has to be done **after** we potentially remove
+    // the marker, as there is no timestamp for removed markers
+    let numberOfHours = 24;
+    if (new Date() - timestamp > numberOfHours * 1000 * 60 * 60) {
+      if (shouldSetState) {
+        // So that react re-renders
+        this._sortMarkers();
+        this.setState({
+          dataSource: this.state.dataSource.cloneWithRows(this.state.markers),
+          loaded: true
+        });
+      }
+      return; // Hide all profiles where the last update was over 1 hour ago
+    }
+
+    if (coordinatesProvided && foundExisting == -1) {
+      let profilePictureUrl = "https://twitter.com/" +
```

Comment:
You got a solid point there!

I thought about it while I was working on the list, but I just didn't have enough time to implement a data store / manager to hold one source of truth for the app.

Another quick workaround would be just passing the users array from the map to the list then display it statically (without live updates).

What do you think?